ethereum/api/src/test-support/java/org/hyperledger/besu/ethereum/api/tls/SelfSignedP12Certificate.java

```diff
@@ -0,0 +1,188 @@
+import java.util.Collection;
+import java.util.Date;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.bouncycastle.asn1.x500.X500Name;
+import org.bouncycastle.asn1.x509.BasicConstraints;
+import org.bouncycastle.asn1.x509.Extension;
+import org.bouncycastle.asn1.x509.GeneralName;
+import org.bouncycastle.asn1.x509.GeneralNames;
+import org.bouncycastle.cert.CertIOException;
+import org.bouncycastle.cert.X509v3CertificateBuilder;
+import org.bouncycastle.cert.jcajce.JcaX509CertificateConverter;
+import org.bouncycastle.cert.jcajce.JcaX509v3CertificateBuilder;
+import org.bouncycastle.jce.provider.BouncyCastleProvider;
+import org.bouncycastle.operator.ContentSigner;
+import org.bouncycastle.operator.OperatorCreationException;
+import org.bouncycastle.operator.jcajce.JcaContentSignerBuilder;
+
+/**
+ * Utility class to generate temporary self-signed certificates in PKCS12 format for testing
+ * purposes using BouncyCastle APIs. The generated certificate supports SAN extension for multiple
+ * DNS and IP addresses
+ *
+ * <p>Note: DO NOT USE IN PRODUCTION. The generated stores and files are marked to be deleted on JVM
+ * exit.
+ */
+public final class SelfSignedP12Certificate {
+  private static final BouncyCastleProvider BOUNCY_CASTLE_PROVIDER = new BouncyCastleProvider();
+  private static final String alias = "test";
+  private static final boolean IS_CA = true;
+  private static final String distinguishedName = "CN=localhost";
+  private static final List<String> sanHostNames = List.of("localhost");
+  private static final List<String> sanIpAddresses = List.of("127.0.0.1");
+  private static final char[] password = "changeit".toCharArray();
+  private final Certificate certificate;
+  private final Path keyStore;
+  private final Path trustStore;
+
+  private SelfSignedP12Certificate(
+      final Certificate certificate, final Path keyStore, final Path trustStore) {
+    this.certificate = certificate;
+    this.keyStore = keyStore;
+    this.trustStore = trustStore;
+  }
+
+  public static SelfSignedP12Certificate create() {
+    try {
+      final KeyPair keyPair = generateKeyPair();
+      final Certificate certificate = generateSelfSignedCertificate(keyPair);
+      final Path keyStore = createKeyStore(keyPair.getPrivate(), certificate);
+      final Path trustStore = createTrustStore(certificate);
+      return new SelfSignedP12Certificate(certificate, keyStore, trustStore);
+    } catch (final IOException | GeneralSecurityException | OperatorCreationException e) {
+      throw new RuntimeException("Error creating self signed certificates", e);
+    }
+  }
+
+  public Certificate getCertificate() {
+    return certificate;
+  }
+
+  public Path getKeyStoreFile() {
+    return keyStore;
+  }
+
+  public Path getTrustStoreFile() {
+    return trustStore;
+  }
+
+  public char[] getPassword() {
+    return password;
+  }
+
+  @SuppressWarnings("DoNotCreateSecureRandomDirectly")
```

Comment:
why are you suppressing - can this not use the secureRandom factory?

Reply:
Since we are in tests and creating not-so-secure certificates, we are ok by suppressing this warning.

Reply:
Please don't suppress this warning, just use `SecureRandomFactory`.

Reply:
👍